src/query/formats/src/field_encoder/values.rs

```diff
@@ -298,14 +308,27 @@ impl FieldEncoderValues {
         in_nested: bool,
     ) {
         let v = unsafe { column.index_unchecked(row_index) };
-        let mut data_cursor = std::io::Cursor::new(v);
-        match Ewkt::from_wkb(&mut data_cursor, WkbDialect::Ewkb) {
-            Ok(ewkt) => self.write_string_inner(ewkt.0.as_bytes(), out_buf, in_nested),
-            Err(_) => {
-                let json = GeoJson(std::str::from_utf8(v).unwrap()).to_json().unwrap();
-                self.write_string_inner(json.as_bytes(), out_buf, in_nested);
-            }
+        let s = match self.common_settings().geometry_format {
+            GeometryDataType::WKB => Ewkb(v.to_vec())
+                .to_wkb(CoordDimensions::xy())
+                .unwrap()
+                .as_bytes()
+                .to_vec(),
+            GeometryDataType::WKT => Ewkb(v.to_vec()).to_wkt().unwrap().as_bytes().to_vec(),
+            GeometryDataType::EWKB => v.to_vec(),
```

Comment:
`EWKB` and `WKB` need to use `hex::encode_upper` to convert binary values to hex format for better display.